src/fractionalintegralmodel.cc

```diff
@@ -0,0 +1,167 @@
+    auto channels = model.get()->get_channels();
+    if(channels.find(*channel) == channels.end()) throw std::invalid_argument(
+        "FractionalGaussianIntegral channel=" + channel->str() + " not in model.get_channels()="
+        + str_iter_refw(channels)
+    );
+}
+
+std::shared_ptr<ProperFractionParameter> FractionalIntegralModel::at(const Channel & channel) {
+    return _data.at(channel);
+}
+std::shared_ptr<const ProperFractionParameter> FractionalIntegralModel::at(const Channel & channel) const {
+    return _data.at(channel);
+}
+
+typename FractionalIntegralModel::Data::iterator FractionalIntegralModel::begin() noexcept {
+    return _data.begin();
+}
+typename FractionalIntegralModel::Data::const_iterator FractionalIntegralModel::cbegin() const noexcept {
+    return _data.begin();
+}
+
+typename FractionalIntegralModel::Data::iterator FractionalIntegralModel::end() noexcept {
+    return _data.end();
+}
+typename FractionalIntegralModel::Data::const_iterator FractionalIntegralModel::cend() const noexcept {
+    return _data.cend();
+}
+
+std::set<std::reference_wrapper<const Channel>> FractionalIntegralModel::get_channels() const {
+    return map_keys_ref_const(_data);
+}
+
+const IntegralModel & FractionalIntegralModel::get_dependent_model() const {
+    return *_model;
+}
+
+inline double _get_integral_remainder(const FractionalIntegralModel * frac, 
+    const IntegralModel & model, const Channel & channel) {
+    return (frac == nullptr ? model.get_integral(channel) : frac->get_integral_remainder(channel));
+}
+
+double FractionalIntegralModel::get_integral(const Channel & channel) const {
+    // TODO: Make this more efficient (conditional is avoidable)
+    return _data.at(channel)->get_value()*_get_integral_remainder(_parent, *_model, channel);
+}
+
+double FractionalIntegralModel::get_integral_remainder(const Channel & channel) const {
+    return (1. - _data.at(channel)->get_value())*_get_integral_remainder(_parent, *_model, channel);
+}
+
+ParamRefs & FractionalIntegralModel::get_parameters(ParamRefs & params, ParamFilter * filter) const {
+    _model->get_parameters(params, filter);
+    for(auto & p: _data) params.push_back(*p.second);
+    return params;
+}
+
+ParamCRefs & FractionalIntegralModel::get_parameters_const(ParamCRefs & params, ParamFilter * filter) const {
+    _model->get_parameters_const(params, filter);
+    for(const auto & p: _data) params.push_back(*p.second);
+    return params;
+}
+
+size_t FractionalIntegralModel::size() const { return _data.size(); }
+
+std::string FractionalIntegralModel::str() const {
+    std::string s = "FractionalIntegralModel({";
+    for(const auto & [channel, integral] : _data) s += channel.get().str() + ": " + integral->str() + ",";
+    return s + "})";
+}
+
+// Return a pointer to a registered FractionalIntegralModel, if it is one
+// TODO: Verify if this is safe, or if dynamic_cast is better
```

Comment:
FWIW, I think `dynamic_cast` would probably be easier to read, and I _suspect_ it'd be at least as efficient as a `std::map` lookup.